webapp/src/lib/components/spot-component/spot-location-create-form.svelte

```diff
@@ -5,100 +5,280 @@
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    export let handleSubmit = (_: Event) => {};
+    export let isVerified: boolean;
 
+    export let houseNumber: number;
     export let streetAddress: string;
     export let city: string;
     export let state: string;
+    export let country: string;
     export let countryCode: string;
     export let postalCode: string;
     export let hasShelter: boolean;
     export let hasPlugIn: boolean;
     export let hasChargingStation: boolean;
 
+    // Province data mapping
     const provinceData: Map<string, string> = new Map([
         ['MB', 'Manitoba'],
         ['ON', 'Ontario'],
         ['AB', 'Alberta'],
         ['QC', 'Quebec'],
         ['NS', 'Nova Scotia'],
         ['BC', 'British Columbia'],
-        ['NL', 'Newfouundland and Labrador'],
+        ['NL', 'Newfoundland and Labrador'],
         ['PE', 'Prince Edward Island'],
         ['SK', 'Saskatchewan'],
         ['YT', 'Yukon'],
         ['NU', 'Nunavut'],
         ['NT', 'Northwest Territories']
     ]);
+
+    // Function to fetch address suggestions
+    const searchAddress = () => {
+        const url = `https://api.geoapify.com/v1/geocode/search?text=${encodeURIComponent(searchQuery)}&apiKey=${apiKey}`;
+
+        fetch(url)
+            .then((response) => response.json())
+            .then((data) => {
+                results = data.features ?? [];
+                dropdownOpen = true;
+            })
+            .catch((error) => {
+                console.error('Error fetching data:', error);
+            });
+    };
+
+    // Handle input changes and debounce the search
+    const handleInput = () => {
+        dropdownOpen = true;
+        isVerified = false; // Reset verification on input change
+        clearTimeout(debounceTimer);
+        debounceTimer = window.setTimeout(() => {
+            if (searchQuery.length > 3) {
+                searchAddress();
+            }
+        }, 500); // 0.5s delay
+    };
+
+    // Handle selection of an address from dropdown
+    const handleSelect = (location: AddressResult) => {
+        searchQuery = location.properties.formatted;
+        results = [];
+        dropdownOpen = false;
+
+        if (location.properties.country_code !== 'ca') {
+            warningModalOpen = true;
+            warningText = 'This application supports locations in Canada only!';
+        } else {
+            houseNumber = location.properties.housenumber ?? '';
+            streetAddress = `${location.properties.housenumber} ${location.properties.street}`;
+            city = location.properties.city;
+            state = location.properties.state_code;
+            countryCode = location.properties.country_code?.toUpperCase();
+            country = location.properties.country;
+            postalCode = location.properties.postcode?.replace(/\s/g, '');
+            isVerified = false; // Reset verification after selection
+            isReadOnly = false;
```

Comment:
You have to update the type in lib/type/address to pass the check/lint here

Reply:
It wouldn't affect it as I am just concatenating two values from it and not changing the structure of it.